server/models/color.model.js

```diff
@@ -1,31 +1,23 @@
-"use strict";
+'use strict';
 
-const { decorate, metadata } = require('../../lib/helpers.js');
-const repository = require("@loopback/repository");
-class Color extends repository.Entity {
-  constructor(data) {
-    super(data);
-  }
-};
+const modelDefinition = {
+  'properties': {
+    // By convetion "id" property should be the id of the model
+    'id': {
+      'type': 'number',
+      'required': true
+    },
+    'value': {
+      'type': 'string',
+      'required': true
+    }
+  },
+  // Supporting these additional features would be a lot of additional work
+  // and some features are still not available in LB4, yet
```

Comment:
Let's forget about LB3 model definition please. In LB4, we have slightly different interface - see [ModelDefinitionSyntax](https://github.com/strongloop/loopback-next/blob/91a37dca4214bb8d0d3cdfbbcf3b64fbd7e7cae2/packages/repository/src/model.ts#L58-L64).

People building models from JavaScript can (and IMO should) use [ModelDefinition class](https://github.com/strongloop/loopback-next/blob/91a37dca4214bb8d0d3cdfbbcf3b64fbd7e7cae2/packages/repository/src/model.ts#L69).

As a general rule I'd like us to follow, our TS APIs should not rely on decorators only, they should always provide a reasonable non-decorator path too. (A side note: decorators are an experimental API, the standardization in TC39 was put on hold. If decorators ever make it into official JavaScript language, their shape may be very different from what we have now.)

LB4 already contains examples on how to create models without decorators, see  https://github.com/strongloop/loopback-next/blob/91a37dca4214bb8d0d3cdfbbcf3b64fbd7e7cae2/packages/repository/test/unit/model/model.unit.ts#L41-L58

The example I pointed to can be easily converted to JavaScript as follows:

```js
  const addressDef = new ModelDefinition('Address');
  addressDef
    .addProperty('street', 'string')
    .addProperty('city', 'string')
    .addProperty('state', String)
    .addProperty('zipCode', STRING);

  class Address extends Entity {
    static definition = addressDef;

    constructor(data?: Partial<Address>) {
      super(data);
    }
  }
```

Personally, I would use the following version:

```js
  class Address extends Entity {
    static definition = new ModelDefinition('Address')
      .addProperty('street', 'string')
      .addProperty('city', 'string')
      .addProperty('state', String)
      .addProperty('zipCode', STRING);

    constructor(data?: Partial<Address>) {
      super(data);
    }
  }
```

**IMPORTANT**

LB2/LB3 users were asking for the ability to define models as regular ES6 classes for a long time. The new LB4 design makes that possible in TypeScript, we should look for ways how to preserve that in JavaScript too.

A typical scenario to consider: how to add a new helper method to a model class.

```js
  class Address extends Entity {
    static definition = new ModelDefinition('Address')
      .addProperty('street', 'string')
      .addProperty('city', 'string')
      .addProperty('state', String)
      .addProperty('zipCode', STRING);

    constructor(data?: Partial<Address>) {
      super(data);
    }

    // Return the address as a multi-line string, e.g. 
    //    1 New Orchard RdArmonk
    //    NY 10504
    getFormattedAddress() 
      return [this.street, this.city, `${this.state} ${this.zipCode}`].join('\n');
    }
  }
```

Reply:
Uh oh, JavaScript does not support property initialization syntax I used above :(


Here are two option how to work around that.

Option 1

```js
const addressDef = new ModelDefinition('Address');
  // add properties, etc.

class Address extends Entity {
  static get definition() { return addressDef; }
}
```

Option 2

```js

class Address extends Entity {
}

Address.definition = new ModelDefinition('Address')
  // add properties, etc.
```